amplify/backend/function/expressServer/src/controllers/leagueController.js

```diff
@@ -0,0 +1,90 @@
+const League = require('../models/league.model')
+
+// @desc create new league. a league is created and sent to the league-data
+// collection in the DB. the user's id is sent through the auth middleware,
+// this id is added to the users array as the first user in the league
+// @route post league/createLeague
+// @access private
+
+const createLeague = async (req, res, next) => {
+    try {
+        // parse the body and destructure
+        const {
+            leagueName,
+            budget,
+            leagueType,
+            tradingFee,
+            portfolios
+          } = req.body
+
+    // check they have sent all fields
+     if (
+      typeof leagueName === 'undefined' ||
+      typeof budget === 'undefined' ||
+      typeof leagueType === 'undefined' ||
+      typeof tradingFee === 'undefined' 
+    ) {
+      // data is missing bad request
+      res.status(400)
+      res.errormessage = 'All details are needed to create a league'
+      return next(
+        new Error(
+          'The client has not sent the required information to create a league',
+        ),
+      )
+    } 
+    let league = await League.findOne({ leagueName })
+    // Check for existing league
+    if (league) {
+      res.status(400)
+      res.errormessage = 'league already exists, need a unique league name'
+      return next(new Error('league already exists'))
+    }
+    // create new league object
+      league = new League({
+        leagueName,
+        budget,
+        leagueType,
+        tradingFee,
+        users: [req.user.id],   // gets this from JWT
+        portfolios: [portfolios]
+      });
+
+    // save league to league-data collection in DB
+    await league.save()
+
+    res.json({league})
+
+    } catch (err) {
+      console.error(err.message);
+      res.errormessage = 'Server error creating a league';
+      return next(err);
+    }
+}
+
+const getLeagues = async (req, res, next) => {
```

Comment:
This a big one, I sent a request with no token and was able to get everyone's leagues which included there emails and passwords. Needs to be a protected route, I should only get the leagues I am associated with, should not return users details. 

Before changing anything on this route, is this the route which will be called when a user clicks into "Leagues" on the frontend is it?

![image](https://user-images.githubusercontent.com/61060096/199701301-4a6f6be8-faa5-4a15-9e11-b1d51b220d3f.png)